File: remappedSrc/com/helixd2s/minevkr/ReflectionUtil.java
package com.helixd2s.minevkr;

import net.minecraft.util.math.Matrix4f;

import java.lang.reflect.Field;
import java.lang.reflect.Modifier;

public class ReflectionUtil {
    public static Field getField(Class clazz, String fieldName) throws NoSuchFieldException {
        try {
            return clazz.getDeclaredField(fieldName);
        } catch (NoSuchFieldException e) {
            Class superClass = clazz.getSuperclass();
            if (superClass == null) {
                throw e;
            } else {
                return getField(superClass, fieldName);
            }
        }
    }
    public static void makeAccessible(Field field) {
        if (!Modifier.isPublic(field.getModifiers()) || !Modifier.isPublic(field.getDeclaringClass().getModifiers()))
        {
            field.setAccessible(true);
        }
    }
    public static float getFieldFValue(Matrix4f obj, String name) throws NoSuchFieldException, IllegalAccessException {
        Field field = ReflectionUtil.getField(obj.getClass(), name);
        ReflectionUtil.makeAccessible(field);
        return field.getFloat(obj);
    }
}

File: src/main/kotlin/com/helixd2s/minevkr/ReflectionUtil.kt
package com.helixd2s.minevkr

import net.minecraft.client.render.RenderLayer

import java.lang.reflect.Field
import java.lang.reflect.Modifier

object ReflectionUtil {
    @Throws(NoSuchFieldException::class)
    fun getField(clazz: Class<*>, fieldName: String?): Field {
        return try {
            clazz.getDeclaredField(fieldName)
        } catch (e: NoSuchFieldException) {
            val superClass = clazz.superclass
            superClass?.let { getField(it, fieldName) } ?: throw e
        }
    }

    fun makeAccessible(field: Field) {
        if (!Modifier.isPublic(field.modifiers) || !Modifier.isPublic(field.declaringClass.modifiers)) {
            field.isAccessible = true
        }
    }

    @Throws(NoSuchFieldException::class, IllegalAccessException::class)
    fun getFieldFValue(obj: Any, name: String?): Float {
        val field = obj?.javaClass?.let { getField(it, name) }
        if (field != null) { makeAccessible(field) }
        if (field != null) { return field.getFloat(obj) }
        return 0.0F
    }

    @Throws(NoSuchFieldException::class, IllegalAccessException::class)
    fun <T> getFieldValue(obj: Any, name: String?): T? {
        val field = obj?.javaClass?.let { getField(it, name) }
        if (field != null) { makeAccessible(field) }
        if (field != null) { return field.get(obj) as T }
        return null
    }
}
